admin/class-manage-fonts.php

```diff
@@ -296,7 +305,82 @@ function ( $font_family ) use ( $font_slug ) {
 			get_stylesheet_directory() . '/theme.json',
 			$theme_json_string
 		);
+	}
+
+	function manage_font_license( $font_name, $file_name ) {
+		if ( ! $font_name ) {
+			return;
+		}
+
+		// Get theme readme.txt
+		$readme_file          = get_stylesheet_directory() . '/readme.txt';
+		$readme_file_contents = file_get_contents( $readme_file );
+
+		if ( ! $readme_file ) {
+			return;
+		}
+
+		// Format font name for font source link
+		$google_formatted_font_name = str_replace( ' ', '+', $font_name );
+
+		// If file_name exists, then add font license to readme.txt
+		if ( 'remove' !== $file_name && is_string( $file_name ) ) {
+			// Check that the font is not already credited in readme.txt
+			if ( false === stripos( $readme_file_contents, $font_name ) ) {
+				// Require php-font-lib
+				require_once( __DIR__ . '/../includes/FontLib/Autoloader.php' );
+
+				// Build font source URL
+				$font_source = 'https://www.google.com/fonts/specimen/' . $google_formatted_font_name;
 
+				// Get license info from font file
+				$font = \FontLib\Font::load( get_stylesheet_directory() . '/assets/fonts/' . $file_name );
+				$font->parse();
+				$license_info = $font->getNameTableString( 13 ) ? $font->getNameTableString( 13 ) : '';
+
+				if ( stripos( $font->getNameTableString( 13 ), 'SIL Open Font License' ) ) {
+					// If license is SIL Open Font License, use custom format
+					$license_info = 'Licensed under the SIL Open Font License, Version 1.1';
+				}
+
+				$license_url = $font->getNameTableString( 14 ) ? "({$font->getNameTableString(14)})" : '';
```

Comment:
Should we include author information to give them credits?

<img width="268" alt="image" src="https://user-images.githubusercontent.com/1935113/231711678-69583778-d00d-4921-ab6d-af16851c3b23.png">

Reply:
Yes this is a good idea, I'll try adding it from the metadata!